main/rendering/pieces.py

```diff
@@ -88,34 +130,75 @@ def render_console(self, renderer):
         self.position = self.next_position
         self.next_position = None
         
-    def render_tkinter(self, renderer):
-        if self.display_image == None:
-            self.load_display_image(renderer)
-    
+    def render_tkinter_with_frame(self, renderer):
+        if self.display_image is None:
+            self.load_display_image_frame(renderer)
+
         next = None
-        if self.next_position == None:
+        if self.next_position is None:
             next = self.position
         else:
             next = self.next_position
+        print(next)
+        if next == vector.Vector2f(-1, -1):
+            self.display_image.grid()
         real_next = self.convert_to_tkinter_coords(next)
         # there might be a better way of doing this
         self.display_image.grid(column=real_next.x + 1, row=real_next.y + 1)
         self.position = next
         self.next_position = None
-    
+
+    def render_tkinter_with_canvas(self, renderer):
+        if self.display_image is not None:
+            next = None
+            if self.next_position is None:
+                next = self.position
+            else:
+                next = self.next_position
+            real_next = self.convert_to_canvas_coords(renderer, next)
+            position_in_canvas_coords = self.convert_to_canvas_coords(renderer, self.position)
+            renderer.canvas.move(self.display_image, -(position_in_canvas_coords.x - real_next.x),
+                                 -(position_in_canvas_coords.y - real_next.y))
+            self.position = next
+            self.next_position = None
+        else:
+            self.load_display_image_canvas(renderer)

```

Comment:
You should invert the condition and add a `return` statement to avoid the `else`